### PR TITLE
[JENKINS-34729] Migrate to 2.9 parent pom

### PR DESCRIPTION
**Pull request by armfergom**
- Migrate to 2.9 parent pom.
- Fix test - Utility method no longer existing.

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -118,27 +113,11 @@
             <artifactId>ini4j</artifactId>
             <version>0.5.2</version>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>matrix-auth</artifactId>
+            <version>1.2</version>
+            <scope>test</scope>
```
Comment (jglick) [inline pom.xml]: Or could use `MockAuthorizationStrategy`.

Comment (jglick) [inline pom.xml]: (Seems to only be used in `doFillCredentialsIdItems`, easily replaced.)

**Comment by jglick** [comment]
:bee: assuming the build passes.

**Comment by armfergom** [comment]
Builds ok locally. Seems like the CI server needs more memory to complete the build successfully.

**Comment by jglick** [comment]
I tried setting a custom `concurrency` on the CI job.

**Comment by armfergom** [comment]
Great. Let's see if that helps. 

In any case, I just want to add one more change to the PR. I just noticed that the plugin does not build ok against 2.x baseline due to `FileCallable` not implementing `checkRoles` method. I'd like to fix that before merging! Sorry for the inconvenience

**Comment by jglick** [comment]
Build 190 did pass. Not sure why the PR status has not been updated.

**Pull request merged**